dev/dll/XamlControlsResources.cpp

```diff
@@ -99,11 +105,38 @@ void XamlControlsResources::UpdateSource()
         }()
     };
 
+    return uri;
+}
+
+void XamlControlsResources::UpdateSource()
+{
+    const bool useNewVisual = UseLatestStyle();
+    const bool areVisualUpdateAPIsAvailable = SharedHelpers::AreVisualUpdateAPIsAvailable();
+
     // Because of Compact, UpdateSource may be executed twice, but there is a bug in XAML and manually clear theme dictionaries here:
-    //  Prior to RS5, when ResourceDictionary.Source property is changed, XAML forgot to clear ThemeDictionaries.
+    // Prior to RS5, when ResourceDictionary.Source property is changed, XAML forgot to clear ThemeDictionaries.
     ThemeDictionaries().Clear();
 
-    Source(uri);
+    try
```

Comment:
Can you explain what is going on here. Why do we need the try/catch block? Can we use conditional compilation here instead?

Reply:
Using conditional compilation now. Thanks.